internal/nix/nix.go

```diff
@@ -154,8 +154,10 @@ func PrintDevEnv(nixShellFilePath, nixFlakesFilePath string) (*varsAndFuncs, err
 		"--extra-experimental-features", "nix-command",
 		"--extra-experimental-features", "ca-derivations",
 		"--option", "experimental-features", "nix-command flakes",
+		"--impure",
 		"--json")
 	debug.Log("Running print-dev-env cmd: %s\n", cmd)
+	cmd.Env = DefaultEnv()
```

Comment:
hmmm is this enough?

what if we're inside a devbox shell and then `devbox add unfree-package`?

Reply:
yup, I tested that scenario as well.

There are two steps when running `devbox add unfree-package`:
1. `nix profile install`: in this PR, we add `--impure` and `cmd.Env = DefaultEnv` to this command.
2. `nix print-dev-env`: this code here (i.e. `internal/nix/nix.go:160`)  will run again with `cmd.Env = DefaultEnv`

Reply:
I'm just curious as to what the diff is for `nix print-dev-env`.. hmm i guess if you don't have the ALLOW_UNFREE flag, then maybe `nix print-dev-env` would fail, because the `flakes.nix` file doesn't have `allowUnfree = 1`